### PR TITLE
CircleCI config: publish new version in pip, update latest version parameter

### PR DESCRIPTION
**Pull request by martinezpl**
This pipeline, upon tag push in whichever branch:

### Job 1:
- spins up container on `cimg/python:3.8` image
- checkouts to the **trigger tag**
- **twine upload**

### Job 2:
- spins up container with aws-cli (our aws environment context applied)
- gets SSM parameter /sdk/latest_versions
- changes python version to **trigger tag**
- updates the SSM parameter

Tested as far as I could without actually publishing or updating the SSM parameter, everything seems fine.

@GetOn4 please look into whether this CI is safe for a public repository

### CODE REVIEW

**Comment by zeritte** [comment]
run publish.sh script isntead of rewriting the steps, easier to manage this way.

**Comment by zeritte** [comment]
manually trigger precommit hook before starting to deploy to ensure code quality

File: .circleci/config.yml [hunk 1]
```diff
@@ -27,8 +35,52 @@ jobs:
...
 
+  pip-publish:
+    docker:
+      - image: cimg/python:3.8
+    environment:
+      TRIGGER_TAG: << pipeline.git.tag >>
+    steps:
+      - run:
+          name: clone main repo, publish tag that triggered the pipeline
+          command: |
```
Comment (GetOn4) [inline .circleci/config.yml:46]: Is there a reason why you don't use the checkout step from circleci?
This is cloning the source code for you via ssh :) 
https://circleci.com/docs/2.0/concepts/

File: .circleci/config.yml [hunk 2]
```diff
@@ -27,8 +35,52 @@ jobs:
...
+            mkdir -p ~/.ssh
+            ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+            git clone git@github.com:aflorithmic/apiaudio-python.git
+            cd apiaudio-python
+            git checkout $TRIGGER_TAG
+            pip3 install twine wheel
+            python3 setup.py sdist bdist_wheel
+            python3 -m twine upload --username $PIPY_USERNAME --password $PIPY_PASSWORD dist/*
```
Comment (GetOn4) [inline .circleci/config.yml:54]: We should include slack notifications for fails/success. I will checkout from this branch and will add it for you. PR incoming

File: .circleci/config.yml [hunk 3]
```diff
@@ -27,8 +35,52 @@ jobs:
...
+      TRIGGER_TAG: << pipeline.git.tag >>
+    steps:
+      - run:
+          name: clone main repo, publish tag that triggered the pipeline
+          command: |
+            mkdir -p ~/.ssh
+            ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+            git clone git@github.com:aflorithmic/apiaudio-python.git
+            cd apiaudio-python
+            git checkout $TRIGGER_TAG
```
Comment (GetOn4) [inline .circleci/config.yml:51]: with checkout step this is automatically :)

File: .circleci/config.yml [hunk 4]
```diff
@@ -27,8 +35,52 @@ jobs:
...
+    executor: aws-cli/default
+    steps:
+      - aws-cli/setup:
+          profile-name: versionUpdate
+      - run:
+          name: update SSM parameter with latest SDK version
+          command: |
+            aws ssm get-parameter --name /sdk/latest_versions > ssm.json
+            cat ssm.json
+            NEW_VALUE=$(cat ssm.json | python3 -c 'import json,sys,os;obj=json.loads(sys.stdin.read());obj=json.loads(obj["Parameter"]["Value"]);obj["python"]=os.environ["TRIGGER_TAG"];print(obj)')
```
Comment (GetOn4) [inline .circleci/config.yml:68]: Do you know JQ command line tool?
jq is like sed for JSON data but python works ofc too

File: .circleci/config.yml [hunk 5]
```diff
@@ -27,8 +35,52 @@ jobs:
...
+            python3 setup.py sdist bdist_wheel
+            python3 -m twine upload --username $PIPY_USERNAME --password $PIPY_PASSWORD dist/*
+
+  update-ssm:
+    environment:
+      TRIGGER_TAG: << pipeline.git.tag >>
+    executor: aws-cli/default
+    steps:
+      - aws-cli/setup:
+          profile-name: versionUpdate
```
Comment (GetOn4) [inline .circleci/config.yml:62]: What's profile-name? Did you create an aws profile?

**Review by GetOn4 — APPROVED** [review]
I don't think we expose any sensible information here and this is safe to go public.

File: .circleci/config.yml (hunk 5 above)
Comment (martinezpl) [reply .circleci/config.yml:62]: It's the profile name in .aws configuration, I think it's juts local and doesn't matter

**Comment by martinezpl** [comment]
> run publish.sh script isntead of rewriting the steps, easier to manage this way.

I can't use the script here. I would need to remove the prompt and modify the twine command anyway @zeritte

**Review by springcoil — APPROVED** [review]
LGTM

**Pull request merged**